### PR TITLE
feat: add BitVec.neg_neg

### PR DESCRIPTION
**Pull request by tobiasgrosser**
.. as well as neg_neq_iff_neq_neg.

### CODE REVIEW

**Comment by tobiasgrosser** [comment]
Draft PR to indicate that @bollu or @alexkeizer should maybe double-check my style choices before Kim does his review.

**Comment by tobiasgrosser** [comment]
I am currently still unhappy with the BitVec.neg_neg proof, as the explicit use of Nat.mod_eq_of_lt is something that comes up very often also in other proofs and is always tedious to write:

```lean4
theorem neg_neg {x : BitVec w} : - - x = x := by
  by_cases h : x = 0
  · simp [h]
  · simp only [toNat_eq, ofNat_eq_ofNat, toNat_ofNat, Nat.zero_mod] at h
    simp only [toNat_eq, toNat_neg]
    rw [Nat.mod_eq_of_lt (a := 2 ^ w - x.toNat) (by omega)]
    rw [Nat.mod_eq_of_lt (by omega)]
    omega
```

**Comment by tobiasgrosser** [comment]
I found a neat way to clean this up with two helper theorems `sub_toNat_mod_cancel` and `sub_sub_toNat_cancel`.

**Comment by tobiasgrosser** [comment]
This is now clean from my perspective and ready to review.

File: src/Init/Data/BitVec/Lemmas.lean [hunk 1]
```diff
@@ -162,6 +162,16 @@ theorem toNat_zero (n : Nat) : (0#n).toNat = 0 := by trivial
 @[simp] theorem toNat_mod_cancel (x : BitVec n) : x.toNat % (2^n) = x.toNat :=
   Nat.mod_eq_of_lt x.isLt
 
+@[simp] theorem sub_toNat_mod_cancel {x : BitVec w} (h : ¬ x = 0#w) :
+    (2 ^ w - x.toNat) % 2 ^ w = 2 ^ w - x.toNat := by
+  simp [bv_toNat] at h
```
Comment (hargoniX) [inline src/Init/Data/BitVec/Lemmas.lean:167]: Non terminal `simp`.

File: src/Init/Data/BitVec/Lemmas.lean [hunk 2]
```diff
@@ -1248,6 +1257,18 @@ theorem neg_eq_not_add (x : BitVec w) : -x = ~~~x + 1 := by
...
+theorem neg_neg {x : BitVec w} : - - x = x := by
+  by_cases h : x = 0#w
+  · simp [h]
+  · simp [bv_toNat, h]
+
+theorem neg_ne_iff_ne_neg {x y : BitVec w} : -x ≠ y ↔ x ≠ -y := by
+  constructor
+    <;> intro h h'
+    <;> subst h'
+    <;> simp [BitVec.neg_neg] at h
```
Comment (hargoniX) [inline src/Init/Data/BitVec/Lemmas.lean:1270]:
```suggestion
theorem neg_ne_iff_ne_neg {x y : BitVec w} : -x ≠ y ↔ x ≠ -y := by
  constructor
  all_goals
    intro h h'
    subst h'
    simp at h
```

**Review by hargoniX — CHANGES_REQUESTED** [review]

File: src/Init/Data/BitVec/Lemmas.lean (hunk 1 above)
Comment (tobiasgrosser) [reply src/Init/Data/BitVec/Lemmas.lean]: Addressed.

**Review by hargoniX — APPROVED** [review]

**Pull request merged**